### PR TITLE
Add tests for MSSQL bracketed identifers

### PR DESCRIPTION
**Pull request by alanwest**
Related to #3574

Adds a couple test cases for MSSQL's bracketed identifiers. This PR just illustrates status quo. When encountering a single quote in a bracketed identifier `db.query.text` and `db.query.summary` are prematurely truncated.

This issue is related to a [discussion](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/3116#discussion_r2349205949) about making our parser overly aggressive and taking a stance to prioritize security over correctness.

I support making improvements to handle these type of scenarios more elegantly so long as we strive not to compromise security. I'd also support moving forward with releasing SqlClient instrumentation without addressing all edge cases. I suspect we'll continue to find new edge cases over time. These could be addressed in future improvements.

### CODE REVIEW

File: test/OpenTelemetry.Contrib.Shared.Tests/SqlProcessorAdditionalTestCases.json [hunk 1]
```diff
@@ -883,5 +883,31 @@
...
+    "input": {
+      "db.system.name": "other_sql",
+      "query": "SELECT [a], [b] FROM [Table'A], [TableB]"
+    },
+    "expected": {
+      "db.query.text": [
+        "SELECT [a], [b] FROM [Table"
+      ],
+      "db.query.summary": "SELECT Table"
+    }
```
Comment (martincostello) [inline test/OpenTelemetry.Contrib.Shared.Tests/SqlProcessorAdditionalTestCases.json:911]: Rather than add a test case that passes when it shouldn't, it might be better to add metadata we can use in the tests to skip it because we know it's broken.

For example, a `"skip": "https://github.com/open-telemetry/opentelemetry-dotnet-contrib/issues/3574"` property?

Then the test can skip itself if the property is present with a non-empty value.

Comment (alanwest) [reply test/OpenTelemetry.Contrib.Shared.Tests/SqlProcessorAdditionalTestCases.json:911]: I'm going to close this PR, but I think a skip property will be useful as we identify other cases we do not yet support.

**Comment by alanwest** [comment]
Closing for now. A solution is in the works. https://github.com/open-telemetry/opentelemetry-dotnet-contrib/issues/3574#issuecomment-3654327722
